profiles/models.py

```diff
@@ -1,3 +1,447 @@
+        choices=Stage.choices,
+        blank=True,
+        null=True
+    )
+
+    def __str__(self):
+        return f"{self.company_name} (Investor, User ID: {self.user_id})"
+
+    class Meta:
+        db_table = "investors"
+        ordering = ["company_name"]
+        verbose_name = "Investor"
+        verbose_name_plural = "Investors"
+        constraints = [
+            models.UniqueConstraint(fields=["user"], name="unique_investor_user")
+        ]
+
+
+class StartupSocialMedia(models.Model):
+    startup = models.ForeignKey(
+        'Startup',
+        on_delete=models.CASCADE,
+        related_name='social_links'
+    )
+    platform = models.CharField(max_length=50)
+    url = models.URLField()
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return f"{self.platform} for {self.startup.company_name}"
+
+    class Meta:
+        db_table = "startup_social_media"
+        ordering = ["-created_at"]
+        verbose_name = "Startup Social Media"
+        verbose_name_plural = "Startup Social Media Links"
+        unique_together = ("startup", "platform")
+
+
+class Interest(models.Model):
+    STATUS_CHOICES = [
+        ('pending', 'Pending'),
+        ('acknowledged', 'Acknowledged'),
+        ('declined', 'Declined')
+    ]
+    investor = models.ForeignKey('Investor', on_delete=models.CASCADE)
+    startup = models.ForeignKey('Startup', on_delete=models.CASCADE)
+    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='pending')
+    message = models.TextField(blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"Interest from {self.investor.company_name} in {self.startup.company_name} - Status: {self.status}"
+
+    class Meta:
+        db_table = "interests"
+        ordering = ["-created_at"]
+        verbose_name = "Investor Interest"
+        verbose_name_plural = "Investor Interests"
+        unique_together = ("investor", "startup")
+
+
+class SavedStartup(models.Model):
+    STATUS_CHOICES = [
+        ('watching', 'Watching'),
+        ('contacted', 'Contacted'),
+        ('negotiating', 'Negotiating'),
+        ('passed', 'Passed')
+    ]
+    investor = models.ForeignKey('Investor', on_delete=models.CASCADE)
+    startup = models.ForeignKey('Startup', on_delete=models.CASCADE)
+    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='watching')
+    notes = models.TextField(blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return f"{self.investor.company_name} saved {self.startup.company_name} ({self.status})"
+
+    class Meta:
+        db_table = "saved_startups"
+        ordering = ["-created_at"]
+        verbose_name = "Saved Startup"
+        verbose_name_plural = "Saved Startups"
+        unique_together = ("investor", "startup")
+
```

Comment:
Do we really need both Interest and SavedStartup? The overlap in purpose (storing status, linking investor to startup) suggests a unified model with a "type" or "category" field could reduce class sprawl.